library/alloc/src/boxed.rs

```diff
@@ -2073,6 +2114,45 @@ impl<I> FromIterator<I> for Box<[I]> {
     }
 }
 
+#[stable(feature = "boxed_slice_into_iter", since = "CURRENT_RUSTC_VERSION")]
+impl<I, A: Allocator> !Iterator for Box<[I], A> {}
+
+// Note: the `#[rustc_skip_during_method_dispatch(boxed_slice)]` on `trait IntoIterator`
+// hides this implementation from explicit `.into_iter()` calls on editions < 2024,
+// so those calls will still resolve to the slice implementation, by reference.
+#[stable(feature = "boxed_slice_into_iter", since = "CURRENT_RUSTC_VERSION")]
+impl<I, A: Allocator> IntoIterator for Box<[I], A> {
+    type IntoIter = vec::IntoIter<I, A>;
```

Comment:
I wonder, should this really be publically the same type? (can't be changed later)

Reply:
It's hard to imagine why we would want something different.

`vec::IntoIter` is
- ptr + cap + alloc (needed to free allocation)
- ptr to first not yielded element
- ptr one past the last not yielded element

You need all the same things for a box. Since the size of initialized elements changes when iterating, there is no difference with a vec -- you need to save the allocation and you need to save the initialized range in it.

Reply:
Also worth pointing out that *not* making them the same type is also something we can't revert after the fact, so, it feels like we should just pick the logical option here and make them the same.